### ISSUE TITLE
Cirm assay uploader

### ISSUE THREAD

**Issue by hongsudt** [informatics-isi-edu/cirm-project#4]
apply the script in informatics-isi-edu/division#33 to upload cirm data.

**Comment by svoinea** [comment]
Done in iobox-win32, commit 2950977a6e293117350e742acd9316d5c2abab47.

**Comment by hongsudt** [comment]
@robes can you review this?

**Comment by robes** [comment]
please update documention

**Comment by robes** [comment]
@svoinea : let's rename the .exe from CirmObserverService.exe to IOboxObserverService.exe

**Comment by carlkesselman** [comment]
I agree.  We should be transiting to a configurable iobox deployment not project specific software.

Carl

Sent from my iPad

---

Dr. Carl Kesselman
Professor, Epstein Department of Industrial and Systems Engineering
Fellow, Information Sciences Institute
Viterbi School of Engineering
University of Southern California

4676 Admiralty Way, Suite 1001, Marina del Rey, CA  90292-6695
Phone:  +1 (310) 448-9338
Email:  carl@isi.edumailto:carl@isi.edu
Web: http://www.isi.edu/~carl

On Apr 15, 2016, at 5:08 PM, robes <notifications@github.com<mailto:notifications@github.com>> wrote:

@svoineahttps://github.com/svoinea : let's rename the .exe from CirmObserverService.exe to IOboxObserverService.exe

##

**Comment by robes** [comment]
@svoinea : i did all the installation steps. you don't say to 'start' the service but i tried anyway. i get a message that says (to paraphrase) the IObox started and then stopped. Not sure how to debug it from here.

**Comment by robes** [comment]
Maybe because I have not setup a config file for it?

**Comment by hongsudt** [comment]
put this back in Todo since @robes is waiting for your answer.

**Comment by robes** [comment]
I created an outbox.conf file in the specified location, again attempted to start the service, but got the same response.

**Comment by svoinea** [comment]
The configuration file contains the key "log" which specifies the log file.
Please check the content of the log file for errors.

**Comment by robes** [comment]
There is no log file in the location that was specified in the config file.

**Comment by robes** [comment]
@svoinea : doc updates needed:
- update the system `PATH` variable with the path to the directory containing the python exe, e.g., `C:\Python27`
- note that the `smtp...` settings can be left out if you want them to be disabled
- note that the log, and workflow directories (input, failure, etc.) must exist
  - better yet we should create these if they dont exist, in my opinion

**Comment by svoinea** [comment]
Renamed CirmObserverService to IOBoxObserverService and all the "CIRM" words by "IOBox".
Updated the README file with the new names.
Fixed bug related to the multi-threading usage.

@robes : First uninstall your current service by running `CirmObserverService.exe -remove"`.
Then pull the latest from Github and follow the instructions from the README (briefly, as the service was renamed, you need to run `IOBoxObserverService.exe -install` to install the service).

**Comment by robes** [comment]
I confirmed that I could uninstall the `Cirm...exe` and install the `IOBox...exe`. Re-configured its properties and it started cleanly. I was able to drop a file in the input dir, for which I had not defined a rule, and confirmed that it correctly moved it to the failure directory and logged the failure reason.

I also noted a couple issues which I added to github. This was a pretty superficial review of installation, configuration, file event handling, and (trivial) failure mode. This is as much time as I have to review this component.
